### PR TITLE
Models/book

### PR DESCRIPTION
**Pull request by juanmbellini**
# Summary

This PR adds the Book model, according to the trello card: [https://trello.com/c/eZNQaOU5](https://trello.com/c/eZNQaOU5)

# Notes

- Replace the ddl-auto property to none, and use [flyway](https://flywaydb.org/) for schema generation, using SQL migration scripts.
- Use the ```orm.xml``` file instead of JPA annotations (this makes a better separation of concerns between the models and the persistence metadata).

The base branch is feature/welcome, which should be changed to dev once [the feature/welcome PR](https://github.com/wolox-training/jmb-java/pull/2) is merged.

# Trello Card

https://trello.com/c/eZNQaOU5

### CODE REVIEW

File: src/main/java/wolox/training/models/Book.java [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+    /**
+     * Asserts the given {@code author}.
+     *
+     * @param author The author value to be asserted.
+     * @throws IllegalArgumentException If the given {@code author} value is not valid.
+     */
+    private static void assertAuthor(final String author) {
+        Assert.notNull(author, "The author must not be null");
+    }
+
```
Comment (darkBasic666) [inline src/main/java/wolox/training/models/Book.java:127]: Why add this asserts? Is for check null before insert in DB ?

File: src/main/resources/META-INF/orm.xml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      </basic>
+      <basic name="pages">
+        <column name="pages" nullable="false" insertable="true" updatable="false"/>
+      </basic>
+      <basic name="isbn">
+        <column name="isbn" nullable="false" insertable="true" updatable="false"/>
+      </basic>
+    </attributes>
+  </entity>
+</entity-mappings>
```
Comment (darkBasic666) [inline src/main/resources/META-INF/orm.xml:46]: Excelent work.

File: src/main/java/wolox/training/models/Book.java [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+            ", author='" + author + '\'' +
+            ", image='" + image + '\'' +
+            ", title='" + title + '\'' +
+            ", subtitle='" + subtitle + '\'' +
+            ", publisher='" + publisher + '\'' +
+            ", year='" + year + '\'' +
+            ", pages=" + pages +
+            ", isbn='" + isbn + '\'' +
+            '}';
+    }
```
Comment (darkBasic666) [inline src/main/java/wolox/training/models/Book.java]: Maybe using [Lombok](https://projectlombok.org/features/Data) @Data reduce ball of text of this class.

File: src/main/resources/META-INF/orm.xml (hunk 2 above)
Comment (juanmbellini) [reply src/main/resources/META-INF/orm.xml:46]: thank you

File: src/main/java/wolox/training/models/Book.java (hunk 3 above)
Comment (juanmbellini) [reply src/main/java/wolox/training/models/Book.java]: Didn't know this project. I'll read about it now

File: src/main/java/wolox/training/models/Book.java (hunk 1 above)
Comment (juanmbellini) [reply src/main/java/wolox/training/models/Book.java:127]: Yes. I prefer the models to validate this stuff themselves, and leave the upper layer (a.k.a the service layer) to validate business stuff (for example, age of a user, and that sort of stuff).

File: src/main/java/wolox/training/models/Book.java (hunk 3 above)
Comment (juanmbellini) [reply src/main/java/wolox/training/models/Book.java]: Okay, replaced stuff with lombok

**Review by Mujics — APPROVED** [review]

**Review by darkBasic666 — APPROVED** [review]

**Pull request merged**